### ISSUE TITLE
`clang-analyzer-alpha.cplusplus.MismatchedIterator` false positive with container insertion

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#132010]
```cpp
#include <list>
#include <unordered_set>

void f()
{
    std::list<int> l;
    std::unordered_set<int> us;
    us.insert(l.cbegin(), l.cend());
}
```

```
<source>:8:5: warning: Container accessed using foreign iterator argument [clang-analyzer-alpha.cplusplus.MismatchedIterator]
    8 |     us.insert(l.cbegin(), l.cend());
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<source>:8:5: note: Container accessed using foreign iterator argument
    8 |     us.insert(l.cbegin(), l.cend());
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
https://godbolt.org/z/arhEMh6Go
